File: week15/week15-1/week15-1.cpp
///LeetCode 3158. Find the XOR of Numbers Which Appear Twice [C ver.]
///тnums柑瞷2Ω计场ノXOR睼癬ㄓ
///计ぶ(程50)ざ1-50ぇ丁┮秨皚穌﹚
///癘计瞷Ω计街瞷2Ω
int duplicateNumbersXOR(int* nums, int numsSize) {
    int table[51] = {}; //皚眖0秨﹍杆1-50ぇ丁计┮秨51
    for(int i=0; i<numsSize; i++){
        table[nums[i]]++; //计瞷Ω计+1
    }
    int ans = 0;
    for(int i=1; i<=50; i++){ //皐癸table柑1-50计
        if(table[i]==2) ans = ans^i;
    }
    return ans;
}

File: week15/week15-1/week15/week15-1/week15-1.py
#Week15-1.py
#LeetCode 3158. Find the XOR of Numbers Which Appear Twice [Python ver.]
class Solution:
    def duplicateNumbersXOR(self, nums: List[int]) -> int:
        table = {} #大括號:table[num] 對應的次數
        for num in nums: #nums 每個數字倫一次
            if num in table: #出現過的話
                table[num] +=1 #次數加1
            else:
                table[num] = 1 #第1次出現
        #print(table) #答案還沒算出來

        ans = 0
        for num in table: #把table裡全部的數字輪一次
            if table[num] ==2: #如果數字剛好出現2次
                ans = ans^num #把答案照題目要求 XOR混進來
        return ans